### PR TITLE
bugfix/async bug

### PR DESCRIPTION
**Pull request by gecBurton**
## Context

In this PR I fix a few bugs:

1. https://github.com/i-dot-ai/redbox/pull/1091#discussion_r1796587381
2. https://github.com/i-dot-ai/redbox/pull/1091#discussion_r1796579465
3. https://github.com/i-dot-ai/redbox/pull/1091#discussion_r1796580363

## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [ ] I have tested any code added or changed
- [x] I have run integration tests https://github.com/i-dot-ai/redbox/actions/runs/11289077170

### CODE REVIEW

File: django_app/redbox_app/redbox_core/consumers.py [hunk 1]
```diff
@@ -216,8 +225,7 @@ def save_message(
     @database_sync_to_async
     def get_ai_settings(chat: Chat) -> AISettings:
         ai_settings = model_to_dict(chat.user.ai_settings, exclude=["label", "chat_backend"])
-        chat_backend = model_to_dict(chat.user.ai_settings.chat_backend)
-        ai_settings["chat_backend"] = chat_backend
+        ai_settings["chat_backend"] = model_to_dict(chat.chat_backend)
```
Comment (gecBurton) [inline django_app/redbox_app/redbox_core/consumers.py:228]: we want to send the `chat_backend` belonging to the current chat not to the `ai_settings`

File: django_app/redbox_app/templates/chats.html [hunk 2]
```diff
@@ -93,7 +93,7 @@ <h3 class="rb-doc-selector__heading govuk-fieldset__heading">Documents to use</h
           <label class="govuk-body-s govuk-!-font-weight-bold" for="llm-selector">Model</label>
           <select id="llm-selector" name="llm" class="govuk-select govuk-!-margin-top-1">
             {% for option in llm_options %}
-              <option value="{{ option.name }}" {% if option.selected %}selected{% endif %}>{{ option.name }}{% if option.default %} (default){% endif %}</option>
+              <option value="{{ option.id }}" {% if option.selected %}selected{% endif %}>{{ option.name }}{% if option.default %} (default){% endif %}</option>
```
Comment (gecBurton) [inline django_app/redbox_app/templates/chats.html:96]: the `ChatLLMBackend` is not uniquely defined by the `name` so we need to use the id instead

**Review by KevinEtchells — APPROVED** [review]
LGTM

**Review by KevinEtchells — APPROVED** [review]

File: tests/test_journey.py [hunk 3]
```diff
@@ -78,7 +78,7 @@ def test_user_journey(page: Page, email_address: str):
     latest_chat_response = chats_page.wait_for_latest_message()
     assert latest_chat_response.text
     # Commented out until we make this visible
```
Comment (KevinEtchells) [inline tests/test_journey.py:80]: I know this is me, but we should probably lose this comment too!

**Pull request merged**